cli/auth_application.go

```diff
@@ -0,0 +1,96 @@
+package cli
+
+import (
+	"encoding/json"
+
+	"github.com/urfave/cli/v2"
+	apppb "go.viam.com/api/app/v1"
+)
+
+// RegisterAuthApplicationAction is the corresponding action for 'third-party-auth-app register'.
+func RegisterAuthApplicationAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.registerAuthApplicationAction(c)
+}
+
+func (c *viamClient) registerAuthApplicationAction(cCtx *cli.Context) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	orgID := cCtx.String(generalFlagOrgID)
+	applicationName := cCtx.String(authApplicationFlagName)
+	originURIs := cCtx.StringSlice(authApplicationFlagOriginURIs)
+	redirectURIs := cCtx.StringSlice(authApplicationFlagRedirectURIs)
+	logoutURI := cCtx.String(authApplicationFlagLogoutURI)
+
+	req := &apppb.RegisterAuthApplicationRequest{
+		OrgId:           orgID,
+		ApplicationName: applicationName,
+		OriginUris:      originURIs,
+		RedirectUris:    redirectURIs,
+		LogoutUri:       logoutURI,
+	}
+	resp, err := c.endUserClient.RegisterAuthApplication(c.c.Context, req)
+	if err != nil {
+		return err
+	}
+
+	infof(cCtx.App.Writer, "Successfully registered auth application")
+	formatOutput, err := json.MarshalIndent(resp, "", "\t")
+	if err != nil {
+		return err
+	}
+	printf(cCtx.App.Writer, "%s", formatOutput)
+	warningf(cCtx.App.Writer, "Keep this information somewhere safe; "+
+		"it contains the secret to your auth application")
```

Comment:
edited this because we're going to add a get command that actually will show the secret again